app/src/split/bluetooth/central.c

```diff
@@ -315,10 +330,30 @@ static struct bt_conn_cb conn_callbacks = {
     .disconnected = split_central_disconnected,
 };
 
+int zmk_split_bt_invoke_behavior(const bt_addr_le_t *source, struct zmk_behavior_binding *binding,
+                                 struct zmk_behavior_binding_event event, bool state) {
+    struct zmk_split_run_behavior_payload payload = {.data = {
+                                                         .param1 = binding->param1,
+                                                         .param2 = binding->param2,
+                                                         .position = event.position,
+                                                         .state = state,
+                                                     }};
+    strncpy(payload.behavior_dev, binding->behavior_dev, ZMK_SPLIT_RUN_BEHAVIOR_DEV_LEN);
```

Comment:
Note that strncpy() does *not* guarantee the copied string is null terminated. As long as the source and destination are the same size (and the source is null terminated) it should be safe, but strlcpy() would be better if that's available.

It's not a standard C library function, but it is a pretty common extension. If it's not available, I found some public domain implementations at https://lwn.net/Articles/33812/

Another quick fix is to strncpy() length - 1 characters and then manually set the last byte to '\0'.

Reply:
Good catch. Fixed with manual last byte setting approach, last I looked `strlcpy` wasn't there.